ws/config.py

```diff
@@ -114,46 +113,46 @@ def setup(self):
 # =======================
 
 def argtype_bool(string):
-    """
-    Convert the string represenation of a bool value to the true
+    """Convert the string represenation of a bool value to the true
     :py:class:`bool` instance.
 
     :returns: ``True`` or ``False``.
     """
     string = string.lower()
-    true_values = {'yes', 'true', 'on', '1'}
-    false_values = {'no', 'false', 'off', '0'}
+    true_values = {"yes", "true", "on", "1"}
+    false_values = {"no", "false", "off", "0"}
     if string in true_values:
         return True
     elif string in false_values:
         return False
     else:
         raise argparse.ArgumentTypeError(
-            "cannot convert '{}'to boolean".format(string))
+            "cannot convert '{}' to boolean".format(string))
 
 
-def argtype_configfile(string):
+def argtype_config(string):
     """Convert `--config` argument to an absolute file path."""
     dirname = os.path.dirname(string)
     name, ext = os.path.splitext(os.path.basename(string))
```

Comment:
You should not split anything. If the config name is `wiki.archlinux.org`, i.e. config path `wiki.archlinux.org.conf`, you would get `.org` in `ext`. Instead, you should just add `.conf` to the name and check if the file exists. Wait, you don't want to do "checks" in argtype functions... Well, you really should!

Reply:
Also at this point it would be good to add some unit tests. Can you look into Pytest and try to add something?

Reply:
I will write tests when all requests will be resolved and the module structure will take its final form.

Reply:
ec89bae 

I left splitting but now the suffix is checked against `.conf` string explicitly. After manual testing in the interactive prompt it looks working properly.